snapshots/snapshot_reporter/src/snapshot_reporter.py

```diff
@@ -0,0 +1,132 @@
+"""
+What do?
+
+"""
+
+import boto3
+from elasticsearch import Elasticsearch
+import httpx
+import humanize
+
+from dateutil import parser
+import json
+import os
+
+
+def get_secret(secret_id):
+    """
+    Use secrets from SecretsManager to construct an Elasticsearch client.
+    """
+    secrets_client = boto3.client("secretsmanager")
+
+    resp = secrets_client.get_secret_value(SecretId=secret_id)
+
+    try:
+        # The secret response may be a JSON string of the form
+        # {"username": "…", "password": "…", "endpoint": "…"}
+        secret = json.loads(resp["SecretString"])
+    except ValueError:
+        secret = resp["SecretString"]
+
+    return secret
+
+
+def get_elastic_client(elastic_secret_id):
+    secret = get_secret(elastic_secret_id)
+
+    return Elasticsearch(
+        secret["endpoint"], http_auth=(secret["username"], secret["password"])
+    )
+
+
+def get_snapshots(es_client, days_to_fetch):
+    index = "snapshots"
+
+    date_filter = [
+        {"range": {"snapshotJob.requestedAt": {"gte": f"now-{days_to_fetch}d/d"}}}
+    ]
+    es_query = {"bool": {"filter": date_filter}}
+    sort_by = [{"snapshotJob.requestedAt": {"order": "desc"}}]
+
+    response = es_client.search(index=index, body={"query": es_query, "sort": sort_by})
```

Comment:
It's a small thing, but I generally prefer to construct the Elasticsearch requests in one big dict, i.e.

```python
request_body = {
    "query": {
        "bool": {
            "filter": {"range": {"snapshotJob.requestedAt": {"gte": f"now-{days_to_fetch}d/d"}}}
        }
    },
    "sort": [{"snapshotJob.requestedAt": {"order": "desc"}}],
}
```

cos then (trailing commas aside) you can drop that straight into the Elastic dev tools for experimenting, and it's easier to follow through to the docs.

Reply:
I find parsing the whole block confusing TBH, but I see the benefit in being able to drop it into the ES console.